pilot/pkg/xds/ads.go

```diff
@@ -1011,18 +1007,14 @@ func (conn *Connection) Watched(typeUrl string) *model.WatchedResource {
 // watched resources for the proxy, ordered in accordance with known push order.
 // It also returns the lis of typeUrls.
 // nolint
-func (conn *Connection) pushDetails() ([]*model.WatchedResource, sets.String) {
+func (conn *Connection) pushDetails() []*model.WatchedResource {
 	conn.proxy.RLock()
 	defer conn.proxy.RUnlock()
-	typeUrls := sets.New[string]()
-	for k := range conn.proxy.WatchedResources {
-		typeUrls.Insert(k)
-	}
-	return orderWatchedResources(conn.proxy.WatchedResources), typeUrls
+	return orderWatchedResources(conn.proxy.WatchedResources)
 }
 
 func orderWatchedResources(resources map[string]*model.WatchedResource) []*model.WatchedResource {
-	wr := make([]*model.WatchedResource, 0, len(resources))
+	wr := make([]*model.WatchedResource, len(resources))
```

Comment:
I'm afraid this change is a bug. Use `append` later, not access the index.

Reply:
You are right, i did not intend to change it. mistake